### DISCUSSION TITLE
RegEx not working in IMaskInput

### DISCUSSION THREAD

**Discussion by easoftwareGit** [uNmAnNeR/imaskjs#1005] [Q&A] [answered]
I cannot get my RegEx mask to work in the IMaskInput. 
The RegEx is 
/^\d(\d)?(,\d(\d)?)*$/
one or two digits, optional comma one or two digits. optional part repeats as needed
It works correctly at [rexex101.com](https://regex101.com/)

I copied the code from the [example](https://imask.js.org/api/modules/react_imask.html), here is my code;

`import { useRef } from "react";
import { IMaskInput } from 'react-imask';

export const TestForm = () => {  
  // use ref to get access to internal "masked = ref.current.maskRef"
  const ref = useRef(null);
  const inputRef = useRef(null);

  return (
	<>
	  ...
	  <IMaskInput
		ref={ref}
		inputRef={inputRef}
		unmask={true}		
		mask={/^\d(\d)?(,\d(\d)?)*$/}
		placeholder="1,4"		
	  />  
	  ...
	</>
  )
} `

In my test, I can enter the first one or two digits, but the comma is never accepted. I have tried all three different values for the unmask prop (true | false | 'typed'), but I get the same results.

Thanks for any help.

**Comment by uNmAnNeR** [comment, marked as answer]
Hi @easoftwareGit,

Your regexp does not match incomplete string like “1,”. But it should match any of intermediate value also. Or you probably can use number mask for it.

**Reply by easoftwareGit** [reply]
Thanks. That did the trick. I did not know I needed to have the RegEx accept an incomplete string.

New RegEx of /\d(\d)?(,(\d(\d)?)?)*$/ works!
